mia/commands/definition.py

```diff
@@ -0,0 +1,396 @@
+    name = None
+    code = None
+
+    package = None
+    if target == 'latest':
+        package = tag.find('package')
+    else:
+        for item in tag.findall('package'):
+            version_code = item.find('versioncode').text
+            if int(version_code) == int(target):
+                package = item
+                break
+
+    if package is not None:
+        name = package.find('apkname').text
+        code = package.find('versioncode').text
+
+    return name, code
+
+
+def download_apps():
+    # Get the MIA handler singleton.
+    handler = MiaHandler()
+
+    # Read the definition settings.
+    settings = handler.get_definition_settings()
+
+    # Read the definition apps lock data.
+    lock_data = handler.get_definition_apps_lock_data()
+
+    # Path where to download the APK files.
+    user_apps_folder = os.path.join(handler.get_definition_path(), 'user-apps')
+    if not os.path.isdir(user_apps_folder):
+        os.makedirs(user_apps_folder, mode=0o755)
+
+    for repo_group in lock_data:
+        print('Downloading %s...' % repo_group)
+        for apk_info in lock_data[repo_group]:
+            print(' - downloading: %s' % apk_info['package_url'])
+            apk_path = os.path.join(user_apps_folder, apk_info['package_name'])
+            path, http_message = urlretrieve(apk_info['package_url'], apk_path)
+            print('   - downloaded %s' %
+                  format_file_size(http_message["Content-Length"]))
+
+    print('Downloading other_apps...')
+    for app_info in settings['other_apps']:
+        print(' - downloading: %s' % app_info['url'])
+        apk_path = os.path.join(user_apps_folder,
+                                os.path.basename(app_info['url']))
+        path, http_message = urlretrieve(app_info['url'], apk_path)
+        print('   - downloaded %s' %
+              format_file_size(http_message["Content-Length"]))
+
+
+def download_os():
+    print('\nNOTE: Command not finished yet; See instructions!\n')
+
+    # Get the MIA handler singleton.
+    handler = MiaHandler()
+
+    # Read the definition settings.
+    settings = handler.get_definition_settings()
+
+    # Create the resources folder.
+    resources_path = os.path.join(handler.get_workspace_path(), 'resources')
+    if not os.path.isdir(resources_path):
+        os.makedirs(resources_path, mode=0o755)
+
+    url = 'https://download.cyanogenmod.org/?device=%s&type=%s' % (
+        settings['general']['cm_device_codename'],
+        settings['general']['cm_release_type']
+    )
+
+    file_name = handler.get_os_zip_filename()
+
+    print("Download CyanogenMod for and save the file as\n - %s\n"
+          "into the resources folder, then verify the file checksum.\n - %s\n"
+          % (file_name, url))
+
+    input_pause('Please follow the instructions before continuing!')
+
+    # Download the CyanogenMod OS.
+    if input_confirm('Extract update binary from the CM zip?', True):
+        extract_update_binary()
```

Comment:
Hm... actually, that file is supposed to suggest stuff that *wasn't* pre-installed, so no worries. It got a little mixed up when mike made most apps pre-installed, but only because he thought MyAppList was broken. (This was before we realized f-droid needed to be manually updated first, to populate the app database.)